### PR TITLE
release post 1.4.0

### PR DESCRIPTION
**Pull request by beta-ziliani**
So far I've only included WASM and the improved inference on ivars/cvars.

@lbguilherme can you double check the instructions here :pray: ?

### CODE REVIEW

**Comment by straight-shoota** [comment]
Other (maybe) notable changes:

* Finalization of Int128 support (I think it's pretty much complete now?)
* LLVM 14
* Enumerable#find! and #index!
* Hash#update
* IO#getb_to_end
* Improvements in interpreter

File: _releases/2022-04-06-1.4.0-released.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```bash
+$ wasm-ld main.wasm -o main-final.wasm -L./lib -lc -lpcre -lrt -lclang_rt.builtins-wasm32
+```
+
+5. Now you have a wasm file that is a final module that can run on any WASI-capable runtime, like Wasmer.
+
+```bash
+$ wasmer main-final.wasm 
+Hello WebAssembly!
+```
```
Comment (lbguilherme) [inline _releases/2022-04-06-1.4.0-released.md:49]: If https://github.com/crystal-lang/crystal/pull/11948 gets merged, then the instructions are:

```md
1. Write a simple Crystal program, let's say: `puts "Hello WebAssembly!"` and name it `main.cr`.
You can also try more complicated programs, but keep in mind that this is still experimental
and not ready for production.
 
2. You need to have `wasm-ld` installed on your system, it is the WebAssembly linker from LLVM.
On some systems, it comes by installing `lld`. Confirm that it is installed by using
`wasm-ld --version`. Please also check that its version is similar to Crystal's LLVM version
(`crystal --version`).

3. You will need `libc` and `libpcre` compiled to the `wasm32-wasi` target. You can fetch a
precompiled version of them here: https://github.com/lbguilherme/wasm-libs/releases.
Check for the `wasm32-wasi-libs.tar.gz` asset, download it and extract it.

4. Compile it with
`CRYSTAL_LIBRARY_PATH=$PWD/wasm32-wasi-libs crystal build main.cr -o main.wasm --target wasm32-wasi`.

5. Run the WebAssembly module with `wasmer main.wasm` or `wasmtime main.wasm` and have fun.
```

If it is not, then step 4 becomes:

```md
4. Cross-compile it with `crystal build main.cr --cross-compile --target wasm32-wasi`.

5. Link it with `wasm-ld main.wasm -o main-final.wasm  -L "$PWD/wasm32-wasi-libs" -lc -lpcre`
```

File: _releases/2022-04-06-1.4.0-released.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @timer_countdown = delay + 10.seconds
+  end
+end
+```
+
+
+## Other notable changes
+
+- Support for LLVM 14 ([PR](https://github.com/crystal-lang/crystal/pull/11905)).
+- Completed support for `Int128` ([PR](https://github.com/crystal-lang/crystal/pull/11576)).
```
Comment (straight-shoota) [inline _releases/2022-04-06-1.4.0-released.md:77]: Well, apparently we're not fully complete yet. There are still a couple of things pending, including some new PRs.

**Review by straight-shoota — COMMENTED** [review]
Some link labels say `PR`, others `details`. That's a bit odd.
The established convention is using the PR number. I'd go with that.

File: _releases/2022-04-06-1.4.0-released.md (hunk 2 above)
Comment (BlobCodes) [reply _releases/2022-04-06-1.4.0-released.md:77]: I'd say the compiler support is now probably finished, but there are still some holes in the stdlib.

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**